site/content/extend/plugins/webapp/interaction-with-redux.md

```diff
@@ -0,0 +1,502 @@
+
+```javascript
+ getChannelByName = (teamId: string, channelName: string, includeDeleted = false) => {
+        return this.doFetch<Channel>(
+            `${this.getTeamRoute(teamId)}/channels/name/${channelName}?include_deleted=${includeDeleted}`,
+            {method: 'get'},
+        );
+    };
+```
+
+**getTeam**
+
+Routes to the team of the specified `teamId` from the Redux store
+
+```javascript
+  getTeam = (teamId: string) => {
+        return this.doFetch<Team>(
+            this.getTeamRoute(teamId),
+            {method: 'get'},
+        );
+    };
+```
+
+**getTeamByName**
+
+Routes to the team of the specified `teamName` from the Redux store
+
+```javascript
+  getTeamByName = (teamName: string) => {
+        this.trackEvent('api', 'api_teams_get_team_by_name');
+
+        return this.doFetch<Team>(
+            this.getTeamNameRoute(teamName),
+            {method: 'get'},
+        );
+    };
+```
+
+**executeCommand**
+
+Executes the specified command with the arguments provided and fetches the response
+
+```javascript
+   executeCommand = (command: string, commandArgs: CommandArgs) => {
+        this.trackEvent('api', 'api_integrations_used');
+
+        return this.doFetch<CommandResponse>(
+            `${this.getCommandsRoute()}/execute`,
+            {method: 'post', body: JSON.stringify({command, ...commandArgs})},
+        );
+    };
+```
+
+**getOptions**
+
+This one is special, because it is needed to make custom requests to the server
+
+```javascript
+  getOptions(options: Options) {
+        const newOptions: Options = {...options};
+
+        const headers: {[x: string]: string} = {
+            [HEADER_REQUESTED_WITH]: 'XMLHttpRequest',
+            ...this.defaultHeaders,
+        };
+
+        if (this.token) {
+            headers[HEADER_AUTH] = `${HEADER_BEARER} ${this.token}`;
+        }
+
+        const csrfToken = this.csrf || this.getCSRFFromCookie();
+        if (options.method && options.method.toLowerCase() !== 'get' && csrfToken) {
+            headers[HEADER_X_CSRF_TOKEN] = csrfToken;
+        }
+
+        if (this.includeCookies) {
+            newOptions.credentials = 'include';
+        }
+
+        if (this.userAgent) {
+            headers[HEADER_USER_AGENT] = this.userAgent;
+        }
+
+        if (newOptions.headers) {
+            Object.assign(headers, newOptions.headers);
+        }
+
+        return {
+            ...newOptions,
+            headers,
+        };
+    }
+```
+
+You can also refer to the [Redux developer guide](https://developers.mattermost.com/contribute/webapp/redux/) to learn more about the [Redux actions](https://developers.mattermost.com/contribute/webapp/redux/actions/), [Redux selectors](https://developers.mattermost.com/contribute/webapp/redux/selectors/) and [Redux reducers](https://developers.mattermost.com/contribute/webapp/redux/reducers/) and gain insights on how these can be used in your Web App plugins.
```

Comment:
I would add before this a `Custom reducers and actions`, stating that you can registerReducer (https://developers.mattermost.com/extend/plugins/webapp/reference/#registerReducer) function.

Reply:
This section is added as well.